packages/patternfly-4/react-core/src/components/Checkbox/Checkbox.js

```diff
@@ -27,16 +34,28 @@ class Checkbox extends React.Component {
   };
 
   render() {
-    const { className, onChange, isValid, isDisabled, ...props } = this.props;
+    const { className, onChange, isValid, isDisabled, label, ...props } = this.props;
     return (
-      <input
-        {...props}
-        className={css(styles.check, className)}
-        type="checkbox"
-        onChange={this.handleChange}
-        aria-invalid={!isValid}
-        disabled={isDisabled}
-      />
+      <div className={css(styles.check, className)}>
+        <input
+          {...props}
+          className={css(styles.checkInput)}
+          type="checkbox"
+          onChange={this.handleChange}
+          aria-invalid={!isValid}
+          disabled={isDisabled}
+        />
+        {React.isValidElement(label)
+          ? label
```

Comment:
this is still an issue in my mind, but maybe @amarie401 or @jschuler can weigh in on if it will cause them any problems. If someone provides
```jsx
import { FormattedMessage } from 'react-intl';

const CustomCheckbox =  (props) => (
  <Checkbox 
    label={<FormattedMessage defaultMessage="Label Value" id="checkbox.label" />}
    {...props}
  />
)
```

The label will go completely unstyled. There is also no way currently to use the label inside this as a component to make sure that I can wrap it any other way. to get around this I would have to do

```jsx
import { injectIntl, defineMessages } from 'react-intl';

const messages = defineMessages({
  label: {
    defaultMessage: 'Label Value',
    id: 'checkbox.label'
  }
})

const CustomCheckbox =  ({ intl, ...props }) => (
  <Checkbox 
    label={intl.formatMessage(messages.label)}
    {...props}
  />
)

const Wrapped = injectIntl(CustomCheckbox);
```

I propose that this return section is just changed to 
```jsx
<label
  className={css(styles.checkLabel, isDisabled && styles.modifiers.disabled)}
  htmlFor={props.id}
>
  {label}
</label>
```
That removes the string and element checks and just passes the node into the label.

Reply:
I agree with Doug here.  I think this leaves room for a label to be unstyled which could lead to an inconsistent user experience.